provider/infoblox.go

```diff
@@ -0,0 +1,381 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	ibclient "github.com/infobloxopen/infoblox-go-client"
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+	"github.com/kubernetes-incubator/external-dns/plan"
+)
+
+// InfobloxProvider implements the DNS provider for Infoblox.
+type InfobloxProvider struct {
+	client       ibclient.IBConnector
+	domainFilter DomainFilter
+	dryRun       bool
+}
+
+// NewInfobloxProvider creates a new Infoblox provider.
+func NewInfobloxProvider(domainFilter DomainFilter, dryRun bool) (*InfobloxProvider, error) {
+	hostConfig := ibclient.HostConfig{
+		Host:     getEnv("GRID_HOST", "127.0.0.1"),
+		Port:     getEnv("WAPI_PORT", "443"),
+		Username: getEnv("WAPI_USERNAME", "admin"),
+		Password: getEnv("WAPI_PASSWORD", ""),
+		Version:  getEnv("WAPI_VERSION", "2.3.1"),
```

Comment:
I would consider moving these configuration options to https://github.com/kubernetes-incubator/external-dns/blob/master/pkg/apis/externaldns/types.go prefixed with `infoblox` and pass them to `NewInfobloxProvider`. This way all configuration options is defined in the same way and visible via `--help` instead of being *hidden* in the provider code.

I know this is not yet the case for all other providers, which I believe is a mistake.